### PR TITLE
Vendor gardener `1.57.1` and upgrade ginkgo to `v2`

### LINKED ISSUE #447: [Feature] Upgrade g/g to version >= `v1.44.0`
**Feature (What you would like to be added):**
I would like a newer version of gardener/gardener (currently: `v1.36.0`) to be used (one that is >= `v1.44.0`)

**Motivation (Why is this needed?):**
g/g versions < `v1.44.0` import `hvpa-controller v0.3.1` which uses `k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible`. Since `etcd-druid` is imported in `gardener` it requires us to have a replace directive for `k8s.io/client-go`.
This upgrade will help us resolve the following https://github.com/gardener/gardener/issues/6807.

**Approach/Hint to the implement solution (optional):**
Code will need to be refactored since [this PR](https://github.com/gardener/gardener/pull/4799) drops `TryUpdate*` which is used in `etcd-druid`. It is mainly used for env tests and can be replaced with standard client methods. Example:
```golang
Eventually(func(g Gomega) {
    g.Expect(c.Get(ctx, client.ObjectKeyFromObject(fullLease), fullLease)).ToSucceed()
    // modify fullLease
    g.Expect(c.Update(ctx, fullLease)).ToSucceed()
}).Should(Succeed())
```

### PR DESCRIPTION
**Pull request by AleksandarSavchev**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
Vendor gardener `v1.57.1`
This PR also upgrades `ginkgo` to `v2` and removes unnecessary replace directives from `go.mod` file.

**Which issue(s) this PR fixes**:
Fixes #447

**Special notes for your reviewer**:
The `ginkgo` upgrade was needed because newer versions of g/g are using `v2` and it was causing problems in some of the env tests

**Release note**:

```other dependency
Dependency `github.com/gardener/gardener` is updated `v1.36.0` -> `v1.57.1`
```
```other dependency
Dependency `github.com/onsi/ginkgo` is upgraded to `github.com/onsi/ginkgo/v2`
```

### CODE REVIEW

File: pkg/mapper/statefulset_to_etcd.go
Comment (aaronfern) [inline pkg/mapper/statefulset_to_etcd.go:38]: These newly added parameters are unused in this function

**Review by aaronfern — COMMENTED** [review]
Thanks for the PR @AleksandarSavchev!

A few questions/comments

File: charts/druid/charts/crds/templates/10-crd-druid.gardener.cloud_etcdcopybackupstasks.yaml
Comment (AleksandarSavchev) [reply charts/druid/charts/crds/templates/10-crd-druid.gardener.cloud_etcdcopybackupstasks.yaml:197]: This field was removed by `make manifests` duo to [this PR](https://github.com/kubernetes-sigs/controller-tools/pull/630) in `controller-tools v0.9.0`

File: pkg/mapper/statefulset_to_etcd.go
Comment (AleksandarSavchev) [reply pkg/mapper/statefulset_to_etcd.go:38]: I have added these parameters since they are needed for the new Mapper interface in order not to change the StatefulSetToEtcd() function. The Mapper interface was changed [in this PR](https://github.com/gardener/gardener/pull/6358).

File: config/rbac/role.yaml
Comment (AleksandarSavchev) [reply config/rbac/role.yaml:12]: These permissions were also removed by running `make manifests`. I am not sure which change caused it. Will have to look into it.

**Review by aaronfern — COMMENTED** [review]
Looks good to me, however a second opinion would be great before I merge this PR
/second-opinion @ishan16696 @abdasgupta

File: controllers/etcd_custodian_controller.go
Comment (plkokanov) [inline controllers/etcd_custodian_controller.go:271]:
```suggestion
		mapper.EnqueueRequestsFrom(druidmapper.StatefulSetToEtcd(ctx, mgr.GetClient()), mapper.UpdateWithNew, c.GetLogger()),
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR! I just have a few minor comments.

File: controllers/etcd_controller.go
Comment (AleksandarSavchev) [reply controllers/etcd_controller.go:456]: https://github.com/gardener/etcd-druid/issues/454

File: Makefile
Comment (aaronfern) [reply Makefile]: I think it should be okay
This role.yaml file generated is used only when deploying druid using kustomize and not in any other case. So should be okay from my pov

**Comment by aaronfern** [comment]
Hi @AleksandarSavchev 
Can you please rebase this PR?

**Review by plkokanov — APPROVED** [review]
/lgtm from my PoV.

**Review by aaronfern — APPROVED** [review]
Thanks for the changes!
/lgtm

**Comment by unmarshall** [comment]
Hi @AleksandarSavchev,
Thanks for the PR. We would like to hold off on this PR for some time till we remove our dependency on /hack/scripts that we source from gardener/gardener into etcd-druid. A lot of dependencies are pulled into the vendor directory when we source `/hack/scripts` and we hardly use any of those dependencies. Once we remove the dependency on g/g we will also upgrade the g/g version to the latest version.

I hope that is ok with you. Apologies for the delay. I was supposed to leave a comment in this issue earlier but was unable to do it. If you have any concerns then please respond here and we can discuss further.

Best Regards,
Madhav

**Comment by shafeeqes** [comment]
Hi, There is another blocker for `gardener/gardener` even after #493, The CRDs generated by etcd-druid are using a very old controller-tools version, and with the `k8s.io/*` `v0.26.0`, they are failing validation.
```
msg: "unable to install CRDs onto control plane: unable to create CRD instances: unable to create CRD \"etcds.druid.gardener.cloud\": CustomResourceDefinition.apiextensions.k8s.io \"etcds.druid.gardener.cloud\" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[backup].properties[compactionResources].properties[claims].items.x-kubernetes-map-type: Invalid value: \"null\": must be atomic as item of a list with x-kubernetes-list-type=set, spec.validation.openAPIV3Schema.properties[spec].properties[backup].properties[resources].properties[claims].items.x-kubernetes-map-type: Invalid value: \"null\": must be atomic as item of a list with x-kubernetes-list-type=set, spec.validation.openAPIV3Schema.properties[spec].properties[etcd].properties[resources].properties[claims].items.x-kubernetes-map-type: Invalid value: \"null\": must be atomic as item of a list with x-kubernetes-list-type=set]",
```
This is probably related to https://github.com/kubernetes/kubernetes/pull/111519.
I tried with the CRDs generated with controller-tools `v0.10.0`, (the ones in this PR), and it passes the validation.

It would be great if we could do something about https://github.com/gardener/etcd-druid/pull/450#issuecomment-1350359260 and get this PR in.
I could also update controller-tools alone in `etcd-druid`, but it also pulls ~1.4L lines, ref: https://github.com/shafeeqes/etcd-druid/tree/test

**Pull request merged**
